### PR TITLE
Location zone and resolver metric support

### PR DESCRIPTION
**Pull request by ampant**
### Proposed changes

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-prometheus-exporter/blob/master/CONTRIBUTING.md) guide
- [ ] I have proven my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have ensured the README is up to date
- [ ] I have rebased my branch onto master
- [ ] I will ensure my PR is targeting the master branch and pulling from my branch on my own fork

### CODE REVIEW

File: collector/nginx_plus.go [hunk 1]
```diff
@@ -364,3 +436,11 @@ func newStreamZoneSyncMetric(namespace string, metricName string, docString stri
 func newStreamZoneSyncZoneMetric(namespace string, metricName string, docString string) *prometheus.Desc {
 	return prometheus.NewDesc(prometheus.BuildFQName(namespace, "stream_zone_sync_zone", metricName), docString, []string{"zone"}, nil)
 }
+
+func newLocationZoneMetric(namespace string, metricName string, docString string, constLabels prometheus.Labels) *prometheus.Desc {
+	return prometheus.NewDesc(prometheus.BuildFQName(namespace, "location_zone", metricName), docString, []string{"resolver"}, constLabels)
+}
+
+func newResolverMetric(namespace string, metricName string, docString string, constLabels prometheus.Labels) *prometheus.Desc {
+	return prometheus.NewDesc(prometheus.BuildFQName(namespace, "location_zone", metricName), docString, []string{"location_zone"}, constLabels)
```
Comment (Dean-Coakley) [inline collector/nginx_plus.go]:
```suggestion
	return prometheus.NewDesc(prometheus.BuildFQName(namespace, "location_zone", metricName), docString, []string{"resolver"}, constLabels)
```

File: collector/nginx_plus.go [hunk 2]
```diff
@@ -110,6 +112,30 @@ func NewNginxPlusCollector(nginxClient *plusclient.NginxClient, namespace string
...
+			"responses_1xx": newLocationZoneMetric(namespace, "responses", "Total responses sent to clients", prometheus.Labels{"code": "1xx"}),
+			"responses_2xx": newLocationZoneMetric(namespace, "responses", "Total responses sent to clients", prometheus.Labels{"code": "2xx"}),
+			"responses_3xx": newLocationZoneMetric(namespace, "responses", "Total responses sent to clients", prometheus.Labels{"code": "3xx"}),
+			"responses_4xx": newLocationZoneMetric(namespace, "responses", "Total responses sent to clients", prometheus.Labels{"code": "4xx"}),
+			"responses_5xx": newLocationZoneMetric(namespace, "responses", "Total responses sent to clients", prometheus.Labels{"code": "5xx"}),
+			"discarded":     newLocationZoneMetric(namespace, "discarded", "Requests completed without sending a response", nil),
+			"received":      newLocationZoneMetric(namespace, "received", "Bytes received from clients", nil),
+			"sent":          newLocationZoneMetric(namespace, "sent", "Bytes sent to clients", nil),
+		},
+		resolverMetrics: map[string]*prometheus.Desc{
```
Comment (Dean-Coakley) [inline collector/nginx_plus.go]: You can find descriptions of the metrics here: https://nginx.org/en/docs/http/ngx_http_api_module.html#def_nginx_resolver_zone

So use those as a guide while following the style of the other metric descriptions.

File: collector/nginx_plus.go [hunk 3]
```diff
@@ -110,6 +112,30 @@ func NewNginxPlusCollector(nginxClient *plusclient.NginxClient, namespace string
...
+			"name":     newResolverMetric(namespace, "name", "desc", prometheus.Labels{"resolver": "name"}),
```
Comment (Dean-Coakley) [inline collector/nginx_plus.go]: I'm not sure what labels we want to use here.

@Rulox thoughts?

File: collector/nginx_plus.go [hunk 4]
```diff
@@ -364,3 +436,11 @@ func newStreamZoneSyncMetric(namespace string, metricName string, docString stri
 func newStreamZoneSyncZoneMetric(namespace string, metricName string, docString string) *prometheus.Desc {
 	return prometheus.NewDesc(prometheus.BuildFQName(namespace, "stream_zone_sync_zone", metricName), docString, []string{"zone"}, nil)
 }
+
+func newLocationZoneMetric(namespace string, metricName string, docString string, constLabels prometheus.Labels) *prometheus.Desc {
+	return prometheus.NewDesc(prometheus.BuildFQName(namespace, "location_zone", metricName), docString, []string{"resolver"}, constLabels)
```
Comment (Dean-Coakley) [inline collector/nginx_plus.go]:
```suggestion
	return prometheus.NewDesc(prometheus.BuildFQName(namespace, "location_zone", metricName), docString, []string{"location_zone"}, constLabels)
```

File: collector/nginx_plus.go (hunk 2 above)
Comment (ampant) [reply collector/nginx_plus.go]: thanks

**Pull request merged**